src/sentry/models/releasefile.py

```diff
@@ -289,10 +299,14 @@ def writable_data(self, create: bool) -> ContextManager[_ArtifactIndexData]:
 
                 target_file.putfile(BytesIO(json.dumps(index_data.data).encode()))
 
-                if not created:
+                artifact_count = index_data.num_files
+                if created:
+                    # NOTE: could pass this to into get_or_create instead
```

Comment:
Since this is a high write area, is it worth doing this?

Reply:
Yes, I'll create a separate PR for this.